### PR TITLE
skip skippable actions if have queue

### PR DESCRIPTION
**Pull request by dqnykamp**

### CODE REVIEW

**Comment by dqnykamp** [comment]
First commit is not part of this PR

File: src/Viewer/PageViewer.jsx [hunk 1]
```diff
@@ -365,41 +377,128 @@ export default function PageViewer(props) {
     baseVariableValue,
     componentName,
     rendererType,
+    promiseResolve,
   }) {
+    // Note: the reason we check both the renderer class and .type
+    // is that if the renderer was memoized, then the renderer class itself in on .type,
```
Comment (jaltekruse) [inline src/Viewer/PageViewer.jsx:383]: possible typo - itself [is] on .type ?

File: src/Viewer/PageViewer.jsx [hunk 2]
```diff
@@ -365,41 +377,128 @@ export default function PageViewer(props) {
...
+        return Promise.resolve(false);
+      }
+    }
 
-      if (baseVariableValue !== undefined && componentName) {
-        updateRendererUpdatesToIgnore({
-          coreId: coreId.current,
-          componentName,
+    if (lastSkippedAction.current) {
+      // we are for sure skipping the lastSkippedAction,
```
Comment (jaltekruse) [inline src/Viewer/PageViewer.jsx:411]: Still working to grok everything going on here, but this language makes it seem like the variable name might be a little misleading. I assume that resolving the promise with false is what is needed to  "really" say it was skipped. So what are the conditions where this "lastSkippedAction" won't be skipped? Might be more awkward, but should this variable be called something like lastSkippableAction?

**Review by jaltekruse — APPROVED** [review]
+1

**Comment by dqnykamp** [comment]
PR is now all commits

File: src/Viewer/PageViewer.jsx (hunk 1 above)
Comment (dqnykamp) [reply src/Viewer/PageViewer.jsx:386]: Thanks!

File: src/Viewer/PageViewer.jsx [hunk 3]
```diff
@@ -527,6 +626,9 @@ export default function PageViewer(props) {
             stateValues,
             childrenInstructions,
             sourceOfUpdate: instruction.sourceOfUpdate,
+            // Note: the reason we check both the renderer class and .type
+            // is that if the renderer was memoized, then the renderer class itself in on .type,
```
[the remark this answers is not in the document]
Comment (dqnykamp) [reply src/Viewer/PageViewer.jsx:633]: Thanks!

File: src/Viewer/PageViewer.jsx (hunk 2 above)
Comment (dqnykamp) [reply src/Viewer/PageViewer.jsx:414]: Right, it is misleading.  I'm renaming it to actionTentativelySkipped, as "last" doesn't make sense in that context given that only one is *tentatively* skipped.

**Pull request merged**